src/EventSubscriber/ConfigSubscriber.php

```diff
@@ -159,4 +159,23 @@ public function onConfigDelete(ConfigCrudEvent $event) {
     }
   }
 
+  /**
+   * Check whether the field encryption config has changed.
+   *
+   * @param \Drupal\Core\Config\Config $config
+   *   The config to check.
+   *
+   * @return bool
+   *   Whether the config has changed.
+   */
+  protected function configChanged(Config $config) {
```

Comment:
The EventSubscriber watches the field_encrypt 3rd party config. If settings change, the encrypted values get re-encrypted, to avoid data loss.
However, the "uncacheable" setting has no impact on the encrypted values, so should be exempt from this check.

Basically this is not really relevant to the caching issue.